ms2rescore/parse_spectra.py

```diff
@@ -26,122 +26,110 @@ def get_missing_values(config, psm_list, missing_rt=False, missing_im=False):
             psm_list_run = PSMList(psm_list=list(chain.from_iterable(psms.values())))
             spectrum_file = infer_spectrum_path(config["spectrum_path"], run)
 
-            rt_dict = None
-            im_dict = None
-
             if missing_im or missing_rt:
                 if spectrum_file.suffix.lower() == ".mzml":
-                    rt_dict, im_dict = _parse_values_from_mzml(
-                        spectrum_file, config, run, missing_rt, missing_im
+                    _parse_values_from_mzml(
+                        psm_list_run, spectrum_file, config, missing_rt, missing_im
                     )
                 elif spectrum_file.suffix.lower() == ".mgf":
-                    rt_dict, im_dict = _parse_values_from_mgf(
-                        spectrum_file, config, run, missing_rt, missing_im
+                    _parse_values_from_mgf(
+                        psm_list_run, spectrum_file, config, missing_rt, missing_im
                     )
                 else:
                     raise MS2RescoreError(
                         f"Could not parse retention time and/or ion mobility from spectrum file for run {run}. "
                         "Please make sure that the spectrum file is either in mzML or MGF format."
                     )
 
-            for value_dict, value in zip([rt_dict, im_dict], ["retention_time", "ion_mobility"]):
-                if value_dict:
-                    try:
-                        psm_list_run[value] = [value_dict[psm.spectrum_id] for psm in psm_list_run]
-                    except KeyError:
-                        raise ParsingError(
-                            f"Could not parse {value} values from spectrum file for run {run}."
-                        )
-
```

Comment:
I'm not sure about the name of this method, `get_missing_values`, given that it doesn't return missing values, but actually fills them with data from the mgf or mzml.